### PR TITLE
cmake,rgw: use jaeger-base target, as a dependency for building dbstore

### PR DESCRIPTION
**Pull request by ideepika**
Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by cbodley — APPROVED** [review]

**Review by rzarzynski — APPROVED** [review]

File: cmake/modules/BuildJaeger.cmake [hunk 1]
```diff
@@ -24,7 +24,7 @@ function(set_library_properties_for_external_project _target _lib)
     INTERFACE_LINK_LIBRARIES "${_libpath}"
     IMPORTED_LINK_INTERFACE_LANGUAGES "CXX"
     IMPORTED_LOCATION "${_libpath}"
-    INTERFACE_INCLUDE_DIRECTORIES "${_includepath}")
+    INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${_includepath}")
```
Comment (tchaikov) [inline cmake/modules/BuildJaeger.cmake:27]: @ideepika what build warning are you trying to suppress ? could you paste a sample of them?

**Comment by ideepika** [comment]
@tchaikov sure! :)
apart from the build logs, it's mostly the unused methods that we do not have definition for in Ceph:
```
13/232] Building CXX object src/common/CMakeFiles/common-common-objs.dir/fs_types.cc[202/232] Building CXX object src/CMakeFiles/common-objs.dir/common/tracer.cc.o
In file included from external/include/opentracing/noop.h:5,
                 from external/include/jaegertracing/Tracer.h:25,
                 from ../src/common/tracer.h:10,
                 from ../src/common/tracer.cc:4:
external/include/opentracing/tracer.h:123:26: warning: 'virtual opentracing::v3::expected<void> opentracing::v3::Tracer::Inject(const opentracing::v3::SpanContext&, const opentracing::v3::CustomCarrierWriter&) const' was hidden [-Woverloaded-virtual]
   virtual expected<void> Inject(const SpanContext& sc,
                          ^~~~~~
In file included from ../src/common/tracer.h:10,
                 from ../src/common/tracer.cc:4:
external/include/jaegertracing/Tracer.h:127:5: warning:   by 'jaegertracing::Tracer::Inject' [-Woverloaded-virtual]
     Inject(const opentracing::SpanContext& ctx,
     ^~~~~~
In file included from external/include/opentracing/noop.h:5,
                 from external/include/jaegertracing/Tracer.h:25,
                 from ../src/common/tracer.h:10,
                 from ../src/common/tracer.cc:4:
external/include/opentracing/tracer.h:145:50: warning: 'virtual opentracing::v3::expected<std::unique_ptr<opentracing::v3::SpanContext> > opentracing::v3::Tracer::Extract(const opentracing::v3::CustomCarrierReader&) const' was hidden [-Woverloaded-virtual]
   virtual expected<std::unique_ptr<SpanContext>> Extract(
                                                  ^~~~~~~
```

**Comment by tchaikov** [comment]
> 'virtual opentracing::v3::expected<void> opentracing::v3::Tracer::Inject(const opentracing::v3::SpanContext&, const opentracing::v3::CustomCarrierWriter&) const' was hidden 

if a method is hidden by a derived class method, can we `using` it in the derived class to expose this method ?

**Comment by ideepika** [comment]
> if a method is hidden by a derived class method, can we `using` it in the derived class to expose this method?

thanks @tchaikov that should be certainly better! testing with updates on shaman

**Comment by ideepika** [comment]
> > if a method is hidden by a derived class method, can we `using` it in the derived class to expose this method?
> 
> thanks @tchaikov that should be certainly better! testing with updates on shaman

@tchaikov i tried using `using` for defining unused methods, but since we are not using class, that doesn't help.. do you think we should be having warnings from external projects, are they good to have from your POV?

**Comment by ideepika** [comment]
https://shaman.ceph.com/builds/ceph/wip-jaeger-only-warnings/50a26a3bbea67d90cec7e5136c1840a4edd3c7d9/

**Comment by yuvalif** [comment]
> > 'virtual opentracing::v3::expected opentracing::v3::Tracer::Inject(const opentracing::v3::SpanContext&, const opentracing::v3::CustomCarrierWriter&) const' was hidden
> 
> if a method is hidden by a derived class method, can we `using` it in the derived class to expose this method ?

@ideepika why did you overloaded a virtual call? this is usually an indication for a design issue

**Comment by ideepika** [comment]
> 
> @ideepika why did you overloaded a virtual call? this is usually an indication for a design issue

@yuvalif we are not, it is a part of jaeger-cpp-client code which is using opentracing code ( both being external projects ) any suggestion are welcome!

**Comment by yuvalif** [comment]
> > @ideepika why did you overloaded a virtual call? this is usually an indication for a design issue
> 
> @yuvalif we are not, it is a part of jaeger-cpp-client code which is using opentracing code ( both being external projects ) any suggestion are welcome!

ok i see, thanks!
probably worthwhile opening an issue in: https://github.com/opentracing/opentracing-cpp/issues
also, I'm not sure how the changes in this PR are suppressing the warning?

**Comment by ideepika** [comment]
> ok i see, thanks! probably worthwhile opening an issue in: https://github.com/opentracing/opentracing-cpp/issues 

sure, I will open an issue.
> also, I'm not sure how the changes in this PR are suppressing the warning?

that is done using including directories with system option: 
`If the SYSTEM option is given, the compiler will be told the directories are meant as system include directories on some platforms. Signalling this setting might achieve effects such as the compiler skipping warnings, or these fixed-install system files not being considered in dependency calculations - see compiler docs.`

changelog:
I have pivoted this PR to just log builds in file.
Fix some rgw missing jaeger target linking found during shaman builds.

opened issue: https://github.com/jaegertracing/jaeger-client-cpp/issues/289

**Review by yuvalif — APPROVED** [review]

**Comment by ideepika** [comment]
https://shaman.ceph.com/repos/ceph/wip-supress-jaeger-cmake-logs/463193bcaf8b3b8a0e616315c8b650acdfe7b53a/

**Pull request merged**